src/index.js

```diff
@@ -34,6 +34,11 @@ class EllipisWithTooltip extends React.Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.children === this.props.children) return;
```

Comment:
it compares the references. if the values are the same do we need to update the state?